collectors/SDRSPropertiesCollector.py

```diff
@@ -0,0 +1,60 @@
+from collectors.PropertiesCollector import PropertiesCollector
+import json
+
+class SDRSPropertiesCollector(PropertiesCollector):
+
+    def __init__(self):
+        super().__init__()
+        self.vrops_entity_name = 'storagepod' 
+        self.label_names = [self.vrops_entity_name, 'vcenter', 'datacenter']
+        self.nested_value_metric_keys = [
+                "config|sdrsconfig|vmStorageAntiAffinityRules"
+                ]
+
+    def get_resource_uuids(self):
+        return self.get_SDRS_clusters_by_target()
+
+    def get_labels(self, resource_id, project_ids):
+        return [self.sdrs_clusters[resource_id]['name'],
+                self.sdrs_clusters[resource_id]['vcenter'],
+                self.sdrs_clusters[resource_id]['parent_dc_name'].lower()] if resource_id in self.sdrs_clusters else []
+
+    def unlock_nested_values(self, statkey, metric_value):
+
+        match statkey:
+            case "config|sdrsconfig|vmStorageAntiAffinityRules":
+                return self.config_sdrsconfig_vmStorageAntiAffinityRules(metric_value)
+
+    def config_sdrsconfig_vmStorageAntiAffinityRules(self, metric_value):
+
+        metric_value = json.loads(metric_value)
+        rules = metric_value.get("rules") if metric_value.get("rules") else []
```

Comment:
```suggestion
        rules = metric_value.get("rules", [])
```

Reply:
That wouldn't work.  The `rules` key is always present and filled with `None`, which cannot be iterated. The following would be useful:
```suggestion
        rules = metric_value.get("rules", []) if metric_value.get("rules") else []
```

Reply:
This will work? It's both `false`, no?

Reply:
Yes this works. If the left hand side is `None`, the right hand side is returned.

```
>>> metric_value
{'rules': None}
>>> foobar = metric_value.get("rules") or []
>>> foobar
[]
>>> metric_value = {"rules": ["a","b","c"]}
>>> foobar = metric_value.get("rules") or []
>>> foobar
['a', 'b', 'c']
```